File: universal/src/main/java/com/github/fernthedev/universal/entity/Coin.java
package com.github.fernthedev.universal.entity;

import com.github.fernthedev.universal.GameObject;
import com.github.fernthedev.universal.EntityID;
import lombok.EqualsAndHashCode;

import java.awt.*;

@EqualsAndHashCode(callSuper = true)
public class Coin extends GameObject {

    public Coin(int x, int y, EntityID entityId) {
        super(x, y, 16, 16, entityId, Color.ORANGE);
        this.velX = 0.0F;
        this.velY = 0.0F;
    }

    protected Coin() {
        super();
    }

    public void tick() {
        this.x += this.velX;
        this.y += this.velY;
    }
}

File: universal/src/main/java/com/github/fernthedev/universal/entity/Coin.kt
package com.github.fernthedev.universal.entity

import com.github.fernthedev.universal.EntityID
import com.github.fernthedev.universal.GameObject
import com.github.fernthedev.universal.Location
import java.awt.Color
import java.util.*


class Coin(location: Location, uniqueId: UUID = UUID.randomUUID()) :
    GameObject(
        location,
        width = 16f,
        height = 16f,
        entityId = EntityID.COIN,
        color = Color.ORANGE,
        velX = 0F,
        velY = 0F,
        uniqueId = uniqueId
    ) {
    init {
        hasTrail = false
    }

    override fun tick() {
    }
}